Build/Scripts/GuardingV2ExtensionsHelper.psm1

```diff
@@ -0,0 +1,229 @@
+        $BaselineVersion = $BaselineVersion + ".0"
+    }
+
+    if (-not ($BaselineVersion -and $BaselineVersion -match "^([0-9]+\.){3}[0-9]+$" )) {
+        throw "Extension Compatibile Version cannot be null or invalid format. Valid format should be like '1.0.2.0'"
+    }
+
+    $appSourceCopJsonPath = Join-Path $ExtensionFolder AppSourceCop.json
+
+    if (!(Test-Path $appSourceCopJsonPath)) {
+        Write-Host "Creating AppSourceCop.json with version $BaselineVersion in path $appSourceCopJsonPath" -ForegroundColor Yellow
+        New-Item $appSourceCopJsonPath -type file
+        $appSourceJson = @{version = '' }
+    }
+    else {
+        $appSourceJson = Get-Content $appSourceCopJsonPath -Raw | ConvertFrom-Json
+    }
+
+    Write-Host "Setting 'version:$BaselineVersion' in AppSourceCop.json" -ForegroundColor Yellow
+    $appSourceJson["version"] = $BaselineVersion
+
+    Write-Host "Setting 'name:$ExtensionName' value in AppSourceCop.json" -ForegroundColor Yellow
+    $appSourceJson["name"] = $ExtensionName
+
+    Write-Host "Setting 'publisher:$Publisher' value in AppSourceCop.json" -ForegroundColor Yellow
+    $appSourceJson["publisher"] = $Publisher
+
+    $buildVersion = Get-ConfigValueFromKey -Key "repoVersion"
+    Write-Host "Setting 'obsoleteTagVersion:$buildVersion' value in AppSourceCop.json" -ForegroundColor Yellow
+    $appSourceJson["obsoleteTagVersion"] = $buildVersion
+
+    # All major versions greater than current but less or equal to main should be allowed
+    $currentBuildVersion = [int] $buildVersion.Split('.')[0]
+    $maxAllowedObsoleteVersion = [int] (Get-ConfigValueFromKey -ConfigType "BuildConfig" -Key "MaxAllowedObsoleteVersion")
+    $obsoleteTagAllowedVersions = @()
+
+    for ($i = $currentBuildVersion + 1; $i -le $maxAllowedObsoleteVersion; $i++) {
+        $obsoleteTagAllowedVersions += "$i.0"
+    }
+
+    Write-Host "Setting 'obsoleteTagAllowedVersions:$obsoleteTagAllowedVersions' value in AppSourceCop.json" -ForegroundColor Yellow
+    $appSourceJson["obsoleteTagAllowedVersions"] = $obsoleteTagAllowedVersions -join ','
+
+    Write-Host "Updating AppSourceCop.json done successfully" -ForegroundColor Green
+    $appSourceJson | ConvertTo-Json | Out-File $appSourceCopJsonPath -Encoding ASCII -Force
+
+    if (-not (Test-Path $appSourceCopJsonPath)) {
+        throw "AppSourceCop.json does not exist in path: $appSourceCopJsonPath"
+    }
+
+    return $appSourceCopJsonPath
+}
+
+<#
+.Synopsis
+    Gets the baseline version for the extension
+.Parameter BuildMode
+    Build mode
+#>
+function Get-BaselineVersion {
+    Param(
+        [Parameter(Mandatory = $true)] 
+        [string] $BuildMode
+    )
+    
+    Import-Module $PSScriptRoot\EnlistmentHelperFunctions.psm1
+
+    if ($BuildMode -eq "Clean") {
+        # Use latest available version from nuget if build mode is clean
+        return (Find-Package -Name "microsoft-ALAppExtensions-Modules-preview" -Source "https://nuget.org/api/v2/").Version
```

Comment:
Just out of curiosity... when we push the package we use source "https://api.nuget.org/v3/index.json". Which one is better or how are they different?

Reply:
Interesting...
I ran it with PS 5.1 and it worked. Could it be fixed? Or it still fails on the runners?